### ISSUE TITLE
Authorization issue with email integration

### ISSUE THREAD

**Issue by AquaOms** [aquasecurity/postee#241]
After configuring the postee integration to email(config file contents below), I am thrown an error.

/# Routes are used to define how to handle an incoming message
routes:
- name: my-email
  outputs: [ my-email ]
  template: vuls-html
......
- name: my-email
  type: email
  enable: true
  user: # Mandatory: SMTP user name (e.g. johndoe@gmail.com)
  password: # Mandatory: SMTP password
  host: 10.26.**.* # Mandatory: SMTP host name (e.g. smtp.gmail.com)
  port: 25 # Mandatory: SMTP server port (e.g. 587)
  sender: ********** # Mandatory: The email address to use as a sender
  recipients: ["************"] # Mandatory: comma separated list of recipients
please note that I have stared the host address, sender and recipient addresses for privacy reasons.

After scanning I am thrown this error: "SendMail Error: smtp: server doesn't support AUTH"
Logs:
2022/01/21 11:17:25 route "my-jira" contains an output "my-jira", which doesn't enable now.
2022/01/21 11:17:25 route "my-email" is associated with template "vuls-html"
2022/01/21 11:17:25 SendMail Error: smtp: server doesn't support AUTH
2022/01/21 11:17:25 From: "***********", to [*********] via "10.26.**.*"
2022/01/21 11:17:25 Error while sending event: smtp: server doesn't support AUTH

I am using an open relay and do not require authorization within my trusted network. 
If possible please add a key AUTH: true/false.

**Comment by simar7** [comment]
Hi @AndreyLevchenko could for take a look at this first? This was reported by a customer so it is higher priority.

**Comment by simar7** [comment]
My initial reaction to this is that you will need to use an App-Specific password for Gmail in this case. AFAIK Google doesn't allow AUTH with basic username/password but instead needs OAuth. 

You can find more details here https://support.google.com/mail/answer/185833#zippy=%2Cwhy-you-may-need-an-app-password%2Cfix-problems-with-microsoft-outlook

Do we currently support OAuth @AndreyLevchenko?

**Comment by simar7** [comment]
In addition @AquaOms - you mentioned that this is within the trusted network. What happens if you omit the username and password fields from the config? From the output it just seems like the server doesn't expect a username/password field.

**Comment by simar7** [comment]
If possible knowing which relay was being used would help as well so we can further read up on the usage of it.

**Comment by AquaOms** [comment]
I am checking, I will get back to you soon. 

FYI - I have opened this issue on behalf of a different user and i am waiting for their response.

**Comment by AquaOms** [comment]
Following is the customers response to the questions:

"I used 10.26.95.5 port 25 open relay. User name and password empty.

I configure postee:
# Routes are used to define how to handle an incoming message
routes:
- name: my-email
  outputs: [ my-email ]
  template: vuls-html
......
- name: my-email
  type: email
  enable: true
  user: # Mandatory: SMTP user name (e.g. johndoe@gmail.com)
  password: # Mandatory: SMTP password
  host: 10.26.95.5 # Mandatory: SMTP host name (e.g. smtp.gmail.com)
  port: 25 # Mandatory: SMTP server port (e.g. 587)
  sender: ******** # Mandatory: The email address to use as a sender
  recipients: ["*********"] # Mandatory: comma separated list of recipients

**Aftel scan i get error: "SendMail Error: smtp: server doesn't support AUTH"**

Logs:
2022/01/21 11:17:25 route "my-jira" contains an output "my-jira", which doesn't enable now.
2022/01/21 11:17:25 route "my-email" is associated with template "vuls-html"
2022/01/21 11:17:25 SendMail Error: smtp: server doesn't support AUTH
2022/01/21 11:17:25 From: "*******", to [********] via "10.26.95.5"
2022/01/21 11:17:25 Error while sending event: smtp: server doesn't support AUTH

**Comment by simar7** [comment]
As discussed offline - I think this is a bug with how Postee handles auth in case of mx server present or otherwise. If a server does not support auth, today the only way to send an email is via using mx-server. The following example could work:

```
- name: my-email
  type: email
  enable: true
  host:      <host> # Mandatopory: SMTP host name (e.g. smtp.gmail.com)
  port:      <port> # Mandatory: SMTP server port (e.g. 587)
  sender:    <sender> # Mandatory: The email address to use as a sender
  recipients: ["<rec>"]  # Mandatory: comma separated list of recipients
  use-mx: true
```

As you can see the user/pass fields are not used. I'm not sure why the logic is written this way but to me it looks like a bug. One should be able to use auth regardless of mx server or not.

@afdesk I see you committed https://github.com/aquasecurity/postee/pull/75 do you have any insight why it was done this way?

**Comment by afdesk** [comment]
@simar7 
PR #75 fixed the issue #60. 
It wasn't a way to use SMTP without the user/pass fields, it was an attempt to use `postee` without SMTP server, to send emails to domain's MX server directly.

**Comment by simar7** [comment]
hi @AquaOms - we've just released https://github.com/aquasecurity/postee/releases/tag/v2.0.8 which should include a couple of fixes for this issue. Let us know by reopening this issue if the problem persists. thanks!
